src/main/java/eu/dissco/core/digitalspecimenprocessor/service/MidsService.java

```diff
@@ -4,38 +4,115 @@
 
+@Slf4j
 @Service
 public class MidsService {
 
+  private static final String MISSING_MESSAGE = "Field does not comply to field: {}";
+
   public int calculateMids(DigitalSpecimen digitalSpecimen) {
     var attributes = getAttributes(digitalSpecimen);
-    if (!compliesTo(MIDSFields.MIDS_1, attributes)) {
+    if (doesNotComplyTo(MIDSFields.MIDS_1, attributes)) {
       return 0;
     }
-    return 1;
+    var midsFields = determineMIDSType(digitalSpecimen);
+    if (midsFields.isEmpty() || doesNotComplyTo(midsFields, attributes)) {
+      return 1;
+    }
+    return 2;
   }
 
-  private boolean compliesTo(List<MIDSFields> fields, JsonNode attributes) {
-    for (var value : fields) {
-      if (attributes.get(value.getTerm()) != null) {
-        var data = attributes.get(value.getTerm()).asText();
-        if (data == null || data.trim().equals("") || data.equals("null")){
-          return false;
+  private List<MIDSFields> determineMIDSType(DigitalSpecimen digitalSpecimen) {
+    var bioType = List.of("BotanySpecimen", "MycologySpecimen", "MicrobiologyMicroOrganismSpecimen",
+        "ZoologyVertebrateSpecimen", "ZoologyInvertebrateSpecimen");
+    var paleoType = List.of("PalaeontologySpecimen");
+    var geoType = List.of("GeologyRockSpecimen", "GeologyMineralSpecimen",
+        "GeologyMixedSolidMatterSpecimen", "AstronomySpecimen");
+    if (bioType.contains(digitalSpecimen.type())) {
+      return MIDSFields.MIDS_2_BIO;
+    } else if (paleoType.contains(digitalSpecimen.type()) || geoType.contains(
+        digitalSpecimen.type())) {
+      return MIDSFields.MIDS_2_GEO_PALEO;
+    } else {
+      log.warn("Digital Specimen has unknown type: {} level 1 is highest achievable",
+          digitalSpecimen.type());
+      return Collections.emptyList();
+    }
+  }
+
+  private boolean doesNotComplyTo(List<MIDSFields> fields, JsonNode attributes) {
+    for (var field : fields) {
+      if (field.equals(MIDSFields.QUANTITATIVE_LOCATION)) {
+        if (isLackingQuantitativeLocation(attributes, field)) {
+          log.debug(MISSING_MESSAGE, field);
+          return true;
+        }
+      } else if (field.equals(MIDSFields.HAS_MEDIA)) {
+        if (isLackingMedia(attributes, field)) {
+          log.debug(MISSING_MESSAGE, field);
+          return true;
         }
       } else {
-        return false;
+        if (isFieldMissing(attributes, field)) {
+          log.debug(MISSING_MESSAGE, field);
+          return true;
+        }
+      }
+    }
+    return false;
+  }
+
+  private boolean isFieldMissing(JsonNode attributes, MIDSFields field) {
+    for (var term : field.getTerm()) {
+      if (attributes.get(term) != null) {
+        var data = attributes.get(term).asText();
+        if (data != null && !data.trim().equals("") && !data.equals("null")) {
+          return false;
+        }
       }
     }
     return true;
   }
 
+  private boolean isLackingMedia(JsonNode attributes, MIDSFields field) {
```

Comment:
In this method, it looks like "field" will always be MIDSFields.HAS_MEDIA. Do you think this will change in the future? If not, there's no need to have it be a method parameter

Reply:
Not sure, it is a bit tricky with the MIDS still a bit up in the air. I will try to rewrite this piece a bit see if there is a cleaner way to do it. Most fields are generic and it just checks if there is any information in one of the terms of the field. However for coordinates they both need to be there (my assumption) and for the hasMedia being there is not enough the value needs to be "`true"` before it complies

Reply:
I was thinking of adding properties to the enum but that also seemed a bit overkill for these specific cases